services/service.go

```diff
@@ -318,24 +331,38 @@ func (s *Service) checkNodeAuthorization(nodeID *models.NodeID, ot creds.Operato
 	return nil
 }
 
-func (s *Service) validateSignedRequest(msg *[]byte, sig *[]byte, ot pb.OperatorType) (*common.Address, error) {
+func (s *Service) validateSignedRequest(msg *[]byte, sig *[]byte, address *common.Address, ot pb.OperatorType) (*common.Address, error) {
 	// Check request age
 	if err := s.checkRequestAge(msg); err != nil {
 		return nil, err
 	}
 
-	// Recover nodeID
+	// First, assume EOA signature
 	nodeID, err := s.getNodeID(msg, sig)
+	if err == nil {
```

Comment:
can we change this to an `if/else` so that we can deduplicate the two calls to `s.checkNodeAuthorization`?

Reply:
Hmm, I don't think I follow how to make that happen.

The call to `getNodeID(msg, sig)` will not give an error if the signature was made with a contract, it will just recover _a random incorrect address_. We do need to call `checkNodeAuthorization` with that address to see if it is actually in the "whitelist".

If I'm missing the point of your request, please keep poking me in the right direction :)

Reply:
hmm yeah maybe not if/else, but
```golang
if err != nil {
  // Attempt EIP-1271 validation
  ...
  valid, err := ...
  if err != nil {
    // return error
  }
}
 // checkNodeAuthz using either the EOA address or SC address
```

Reply:
But how would we know to check with either the EOA address or the SC address? We need to check with _both_. Either neither of them will pass `checkNodeAuthz`, or one of them will pass `checkNodeAuth`. AFAIK it's not possible to know which one is "valid" until we do the actual check.

Reply:
As mentioned offline- checking the EOA sig needs to return an error if the recovered address doesn't match the one in the json payload. I think it's possible for getNodeID to return a valid address for an EIP1271 signature, just essentially random bytes.

Once we have that validation, we know a priori whether an EOA signed the message, and we can implement this as a more "pure" bit of fallback logic (ie, first check EOA signature, if not valid, check EIP1271, and use the appropriate address in a single checkNodeAuthorization call.)

Reply:
Will close https://github.com/Rocket-Rescue-Node/rescue-api/issues/26